cheroot/server.py

```diff
@@ -1660,26 +1660,17 @@ def prepare(self):
 
         # Select the appropriate socket
         self.socket = None
+        msg = 'No socket could be created'
         if os.getenv('LISTEN_PID', None):
             # systemd socket activation
             self.socket = socket.fromfd(3, socket.AF_INET, socket.SOCK_STREAM)
         elif isinstance(self.bind_addr, six.string_types):
             # AF_UNIX socket
-
-            # So we can reuse the socket...
-            try:
-                os.unlink(self.bind_addr)
-            except Exception:
-                pass
-
-            # So everyone can access the socket...
             try:
-                os.chmod(self.bind_addr, 0o777)
-            except Exception:
-                pass
-
-            info = [
-                (socket.AF_UNIX, socket.SOCK_STREAM, 0, '', self.bind_addr)]
+                self.bind_unix_socket(self.bind_addr)
+            except socket.error as serr:
```

Comment:
This error handling is asymmetric to how a socket error is handled for AF_INET*.

Reply:
What do you mean? Moving `socket.close()` inside the method? I hope to generalize this stuff sometime in future.